src/ir_passes/assignment_check.rs

```diff
@@ -0,0 +1,110 @@
+use std::collections::HashMap;
+
+use codespan_reporting::{
+    diagnostic::Diagnostic,
+    term::{
+        self,
+        termcolor::{ColorChoice, StandardStream},
+    },
+};
+use itertools::Itertools;
+
+use crate::{
+    ir::{Component, Connect, Ctx, DisplayCtx, PortIdx},
+    ir_visitor::{Action, Visitor},
+    utils::{GPosIdx, GlobalPositionTable},
+};
+
+#[derive(Default)]
+/// Makes sure each index in a port is only written to at most once
+/// Must occur after monomorphization.
```

Comment:
This should also ensure that there is at least one write. We need to make sure that there is exactly one write to each location of the bundle.

Reply:
Should we make 0 writes a warning or an error? It seems less severe of a problem than multiple writes, but I'm not sure.

Reply:
If you read a value from a location but no one writes to it, what is the inlining supposed to do?

I think it should be an error